### PR TITLE
Recover deadlock

### PR DESCRIPTION
**Pull request by ldennis**
- Verify table version does not change after recover_deadlock()
- tableversion remains unchanged for rebuild and truncate
- update db struct and bdb_state in place for alter/rebuld/truncate
- refactor truncate code to behave like alter instead of drop+add

### CODE REVIEW

File: db/sqlinterfaces.c [hunk 1]
```diff
@@ -5296,8 +5300,8 @@ int sqlserver2sqlclient_error(int rc)
 {
     switch (rc) {
     case SQLITE_DEADLOCK:
-        return CDB2ERR_DEADLOCK;
     case SQLITE_BUSY:
+    case SQLITE_COMDB2SCHEMA:
         return CDB2ERR_DEADLOCK;
```
Comment (ldennis) [inline db/sqlinterfaces.c]: make it cdb2err_schema

File: db/sqloffload.c [hunk 2]
```diff
@@ -742,6 +752,8 @@ static void osql_scdone_commit_callback(struct ireq *iq)
             broadcast_sc_end(iq->sc->table, iq->sc_seed);
             if (iq->sc->db)
                 sc_del_unused_files(iq->sc->db);
+            if (iq->sc->fastinit && !iq->sc->drop_table)
```
Comment (dorinhogea) [inline db/sqloffload.c]: why do we run analyze after fastinit?

**Comment by dorinhogea** [comment]
Looks good to me.

File: db/sqloffload.c (hunk 2 above)
Comment (ldennis) [reply db/sqloffload.c]: To clear sqlite_stats

**Pull request merged**